### PR TITLE
Macros: Add section on metadata introspection

### PR DESCRIPTION
**Pull request by jakemac53**
This attempts to address the major concerns around the metadata introspection API.

I chose to be pretty flexible here, and provide essentially a "best effort" api, that may fail when called if the annotation cannot be instantiated yet.

### CODE REVIEW

**Review by munificent — COMMENTED** [review]
Do you still want me to review this in light of our discussions last week around metadata annotation arguments always being provided to macros as metaobjects and not objects? I'm happy to review if it's still useful or help out rewriting this to be more... meta-y. :)

**Comment by jakemac53** [comment]
> Do you still want me to review this in light of our discussions last week around metadata annotation arguments always being provided to macros as metaobjects and not objects? I'm happy to review if it's still useful or help out rewriting this to be more... meta-y. :)

That is actually the api I propose in this section - although I should probably move that up (it is the last sub-section right now).

File: working/macros/feature-specification.md [hunk 1]
```diff
@@ -292,12 +292,86 @@ The [ClassDeclaration][] instance you get here provides all the
...
+The `builder` parameter also provides an API that allows you to retrieve an
 introspection object for any `Type` object available to the macro at runtime.
 The introspection capabilites of these objects are limited to the information
 produced by the previous macro phase of macros, similar to the capabilites
 provided for type references on the declaration.
 
+### Metadata Annotations
+
+The ability to introspect on metadata annotations is important for macros, as
+it is expected to be a common way to configure them. For instance a class level
```
[the remark this answers is not in the document]
Comment (jakemac53) [reply working/macros/feature-specification.md:304]: I do think we want to actually give macros the evaluated constant for the annotation, and not a reflection api over the arguments themselves. Possibly we actually need a more general constant evaluation api? But I will reword this to remove the "configuration" terminology for now.
